lib/client/base.js

```diff
@@ -745,9 +745,16 @@
   HappnClient.prototype.reconnect = function (options) {
     var _this = this;
 
+    if (!this.reconnectAttempt) this.reconnectAttempt = 0;
+    var reconnectAttempt = ++this.reconnectAttempt;
+
     _this.authenticate(function (e) {
+      if (_this.reconnectAttempt != reconnectAttempt) return; // this is not the last attempt, ignore it
```

Comment:
The issue is more that a previous one was still timing out, so we might have reconnected, but we reconnect again.

I think I've resolved it cleaner now.